packages/firestore/src/local/local_store.ts

```diff
@@ -642,6 +628,84 @@ export class LocalStore {
       });
   }
 
+  /**
+   * Prepares document change buffer resulted from documents to be applied,
+   * returns the document changes resulting from applying those documents.
+   *
+   * @param txn Transaction to use to read existing documents from storage.
+   * @param documentBuffer Document buffer to collect the resulted changes to be
+   *        applied to storage.
+   * @param documents Documents to be applied.
+   * @param globalVersion A `SnapshotVersion` representing the read time if all
+   *        documents have the same read time.
+   * @param documentVersions A DocumentKey-to-SnapshotVersion map if documents
+   *        have their own read time.
+   *
+   * Note: this function will use `documentVersions` if it is defined;
+   * when it is not defined, resorts to `globalVersion`.
+   */
+  prepareApplyingDocuments(
```

Comment:
s/prepareApplyingDocuments/prepareToApplyDocuments/ (or back to `applyDocuments`)

Or rather: `populateDocumentChangeBuffer`?

Reply:
`populateDocumentChangeBuffer` sounds good.